lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -200,6 +200,62 @@ defmodule ExUnit.DocTestTest.IndentationNotEnough do
   def test_fun, do: :ok
 end |> write_beam
 
+defmodule ExUnit.DocTestTest.FencedHeredocs do
+  @doc ~S'''
+  Receives a test and formats its failure.
+
+  ## Examples
+
+  ```
+  iex> 1 + 2
+  3
+  ```
+
+      ```
+      iex> 1 + 2
+      3
+      ```
+
+  ```
+      iex> 1 + 2
+      3
+  ```
+  '''
+  def heredocs, do: :ok
+end |> write_beam
+
+defmodule ExUnit.DocTestTest.FencedIndentationMismatchedPrompt do
```

Comment:
I've figured out a couple of issue with these tests and put them in the correct places throughout the suite such that the tests pass per the current implementation.

Whether or not that's the desired behaviour is up to you. In summary:

- Fenced code blocks mostly just work, even if non-terminated.
- However, the doctests will raise a `SyntaxError` if:
  - The fence indentation is mismatched
  - The fence content is dedented further than the fence